src/firefly/java/edu/caltech/ipac/util/FileUtil.java

```diff
@@ -1013,10 +1012,107 @@ private static URL getThisClassURL() {
     }
 
     private static String getThisClassFileName() {
-        String cName= FileUtil.class.getName();
+        String cName = FileUtil.class.getName();
         return cName.replace(".", "/") + ".class";
     }
 
+    /**
+     * Determines the MIME type of the given file.
+     * This method may return an incorrect MIME type if the file extension is registered,
+     * and no other detection method is available.
+     *
+     * @param file input file to detect
+     * @return A String representing the MIME type of the file, or "unknown" otherwise
+     */
+    public static String getMimeType(String file) {
+        return new Tika().detect(file);
+    }
+
+    /**
+     * Detects the format of the given file
+     *
+     * @param fname The full path of the file to detect the format for.
+     * @return The detected {@code Format} of the file, or {@code UNKNOWN} if the format could not be determined.
+     * @throws IOException If an I/O error occurs while accessing the file.
+     */
+    @Nonnull
+    public static Format detectFormat(String fname) throws IOException {
```

Comment:
I am not sure this `detectFormat` function. It probably needs to need in its own file. It make FileUtil.java have a lot of external dependencies. Before it was only dependent on java.